roles/openshift_health_checker/openshift_checks/__init__.py

```diff
@@ -94,7 +120,30 @@ def subclasses(cls):
             for subclass in subclass.subclasses():
                 yield subclass
 
-    def execute_module(self, module_name=None, module_args=None):
+    def register_failure(self, error):
+        """Record failure in the check, to be merged into result hash and saved to output directory."""
+        # It should be an exception; make it one if not
+        if not isinstance(error, OpenShiftCheckException):
+            error = OpenShiftCheckException(str(error))
+        self.failures.append(error)
+        # duplicate it in the notes so it can be seen in the context of any
+        # information that led to the failure
+        self.register_note("failure: " + error.name, str(error))
+
+    def register_note(self, context, msg):
+        """Record a note for the check, intended to be saved as an ordered list in a local notes file."""
```

Comment:
What is the purpose of a note? I think that explanation would be useful to have in the docstring.